examples/hello-world.rs

```diff
@@ -9,158 +9,220 @@ use wgpu::{
     RenderPassDescriptor, RequestAdapterOptions, SurfaceConfiguration, TextureFormat,
     TextureUsages, TextureViewDescriptor,
 };
-use winit::{
-    dpi::LogicalSize,
-    event::{Event, WindowEvent},
-    event_loop::EventLoop,
-    window::WindowBuilder,
-};
+use winit::{dpi::LogicalSize, event::WindowEvent, event_loop::EventLoop, window::Window};
 
 fn main() {
-    pollster::block_on(run());
+    let event_loop = EventLoop::new().unwrap();
+    event_loop
+        .run_app(&mut Application { window_state: None })
+        .unwrap();
 }
 
-async fn run() {
-    // Set up window
-    let (width, height) = (800, 600);
-    let event_loop = EventLoop::new().unwrap();
-    let window = Arc::new(
-        WindowBuilder::new()
+struct WindowState {
+    device: wgpu::Device,
+    queue: wgpu::Queue,
+    surface: wgpu::Surface<'static>,
+    surface_config: SurfaceConfiguration,
+
+    font_system: FontSystem,
+    swash_cache: SwashCache,
+    _cache: glyphon::Cache,
```

Comment:
It looks like some of these fields might not need to be stored as part of `WindowState` if I'm understanding correctly, could we remove any that we don't need to keep around?

Reply:
Alright. The only one I could omit was `_cache: glyphon::Cache`.